File: pyg_manager/pyg_manager_interface/src/main/java/com/huyaaaaaa/manager/service/SpecificationService.java
package com.huyaaaaaa.manager.service;

import com.huyaaaaaa.pojo.TbSpecification;
import com.huyaaaaaa.utils.PageResult;
import com.huyaaaaaa.utils.PygResult;
import com.huyaaaaaa.vo.Specification;

import java.util.List;

public interface SpecificationService {
    List<TbSpecification> findAll();
    PageResult<TbSpecification> findPage(Integer pageNum, Integer pageSize);
    PygResult add(Specification specification);

    TbSpecification findOne(Long id);


    PygResult update(Specification specification);

    PygResult delete(Long[] ids);

    PageResult<TbSpecification> search(Integer page, Integer rows, String searchStr);
}

File: pyg_manager_web/src/main/webapp/js/service/specificationService.js
app.service("specificationService",

    function($http){

        //查询所有
        this.findAll=function(){
            return $http.get("../specification/findAll");
        }

        //查询单个
        this.findOne=function(id){
            return $http.get("../specification/findOne?id=" + id);
        }

        //分页查（已废弃）
        this.findPage=function(page,rows){
            return $http.get("../specification/findPage?page=" + page + "&rows=" + rows);
        }

        //条件分页查询
        this.search=function(page,rows,searchStr){
            return   $http.post("../specification/search?page=" + page + "&rows=" + rows,searchStr);
        }

        //添加单个
        this.add=function(items){

            return $http.post("../specification/add",items);
        }

        //修改单个
        this.update=function(items){
            return $http.post("../specification/update",items);
        }

        //删除操作
        this.dele=function(selectedItems){
            return $http.get("../specification/delete?ids="+selectedItems);
        }
        this.selectSpecMapList=function () {
            return $http.get("../specification/selectSpecMapList")
        }
    })